src/d3a/models/strategy/area_agents/two_sided_agent.py

```diff
@@ -15,35 +15,41 @@
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """
+from d3a_interface.constants_limits import ConstSettings
 from numpy.random import random
+
 from d3a.models.strategy.area_agents.one_sided_agent import OneSidedAgent
 from d3a.models.strategy.area_agents.two_sided_engine import TwoSidedEngine
-from d3a_interface.constants_limits import ConstSettings
 
 
 class TwoSidedAgent(OneSidedAgent):
+    """Handles order forwarding between two sided markets."""
 
     def __init__(self, *, owner, higher_market, lower_market,
                  min_offer_age=ConstSettings.IAASettings.MIN_OFFER_AGE,
-                 min_bid_age=ConstSettings.IAASettings.MIN_BID_AGE):
-        self.engines = [
-            TwoSidedEngine('High -> Low', higher_market, lower_market, min_offer_age,
-                           min_bid_age, self),
-            TwoSidedEngine('Low -> High', lower_market, higher_market, min_offer_age,
-                           min_bid_age, self),
-        ]
+                 min_bid_age=ConstSettings.IAASettings.MIN_BID_AGE,
+                 do_create_engine=True):
+        if do_create_engine:
+            self.engines = [
+                TwoSidedEngine("High -> Low", higher_market, lower_market, min_offer_age,
+                               min_bid_age, self),
+                TwoSidedEngine("Low -> High", lower_market, higher_market, min_offer_age,
+                               min_bid_age, self),
+            ]
         super().__init__(owner=owner,
                          higher_market=higher_market, lower_market=lower_market,
                          min_offer_age=min_offer_age, do_create_engine=False)
 
     def usable_bid(self, bid):
-        """Prevent IAAEngines from trading their counterpart's bids"""
+        """Prevent IAAEngines from trading their counterpart's bids."""
         return all(bid.id not in engine.forwarded_bids.keys() for engine in self.engines)
 
+    # pylint: disable=unused-argument
```

Comment:
I am frankly surprised that this error is raised in some cases and not others (for instance, in other cases pylint can detect that the base class is BaseStrategy that implements the same methods with the same argument names, therefore it can detect that you do it that way here in order to adhere to the base class interface). Nevertheless nothing to do here unfortunately .